src/AppBundle/Controller/CategoryController.php

```diff
@@ -31,8 +31,26 @@ public function categoryDetail(Request $request)
 			throw new NotFoundHttpException("Kategorie neexistuje");
 		}
 
+		// paginator
+        $em = $this->get('doctrine.orm.entity_manager');
+        $dql = 'SELECT p
```

Comment:
Jo souhlas. Když je podíváš na ukázkové řešní v masteru, tak podobný problém jsem tam řešil já. Může to být ukázkové řešení. Nicméně nelíbí se mi, protože je tam porušený jeden z pricipů SOLID. Ale to vysvětlím.